mixer/adapter/kubernetesenv/kubernetesenv_norace_test.go

```diff
@@ -0,0 +1,65 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kubernetesenv
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/pkg/adapter/test"
+	"istio.io/istio/pkg/kube/secretcontroller"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+// This test is skipped by the build tag !race due to https://github.com/istio/istio/issues/15610
+func Test_KubeSecretController(t *testing.T) {
```

Comment:
this is just moved so I didn't disable all the other tests in the same file, I didn't change anything here

Reply:
Does this test fail only on racetests, or just generally?  It could be the case that changes to this code to prevent controller leaks are breaking the functionality of this test.

Reply:
AFAIK it only fails on race tests. Before moving this I verified it still fails on race tests. Its not 100% of the time, more like 5% or so.